### PR TITLE
[SES-637] - Stop playing message if deleted

### PR DESCRIPTION
**Pull request by simophin**
### Contributor checklist

- [ ] I have tested my contribution on these devices:
 * Device A, Android X.Y.Z
 * Device B, Android Z.Y
 * Virtual device W, Android Y.Y.Z
- [x] My contribution is fully baked and ready to be merged as is
- [ ] I ensure that all the open issues my contribution fixes are mentioned in the commit message of my first commit using the `Fixes #1234` [syntax](https://help.github.com/articles/closing-issues-via-commit-messages/)

----------

### CODE REVIEW

**Review by bemusementpark — APPROVED** [review]

File: app/src/main/java/org/thoughtcrime/securesms/conversation/v2/ConversationViewModel.kt [hunk 1]
```diff
@@ -103,6 +104,13 @@ class ConversationViewModel(
         }
     }
 
+    override fun onCleared() {
+        super.onCleared()
+
+        // Stop all voice message when existing the convo page
```
[the remark this answers is not in the document]
Comment (simophin) [reply app/src/main/java/org/thoughtcrime/securesms/conversation/v2/ConversationViewModel.kt:110]: Good catch!

File: app/src/main/java/org/thoughtcrime/securesms/conversation/v2/ConversationViewModel.kt [hunk 2]
```diff
@@ -142,10 +150,24 @@ class ConversationViewModel(
...
+     * Stops audio player if its current playing is the one given in the message.
+     */
+    private fun stopPlayingAudioMessage(message: MessageRecord) {
+        val player = AudioSlidePlayer.getInstance()
+        val audioSlide = player?.audioSlide
+        if (audioSlide != null &&
+                message is MmsMessageRecord &&
+                message.slideDeck.audioSlide == audioSlide) {
+            player.stop()
+        }
```
[the remark this answers is not in the document]
Comment (simophin) [reply app/src/main/java/org/thoughtcrime/securesms/conversation/v2/ConversationViewModel.kt:168]: Sure let's do that

File: app/src/main/java/org/thoughtcrime/securesms/conversation/v2/ConversationActivityV2.kt [hunk 3]
```diff
@@ -103,6 +103,7 @@ import org.thoughtcrime.securesms.ApplicationContext
 import org.thoughtcrime.securesms.PassphraseRequiredActionBarActivity
 import org.thoughtcrime.securesms.attachments.ScreenshotObserver
 import org.thoughtcrime.securesms.audio.AudioRecorder
+import org.thoughtcrime.securesms.audio.AudioSlidePlayer
```
[the remark this answers is not in the document]
Comment (simophin) [reply app/src/main/java/org/thoughtcrime/securesms/conversation/v2/ConversationActivityV2.kt]: Yep. We should def put in some automation for the code standard :-)

**Review by bemusementpark — APPROVED** [review]

**Review by AL-Session — APPROVED** [review]
All looks fine to me.

**Review by bemusementpark — APPROVED** [review]

**Pull request merged**